File: ackermann.c
// gcc -O3 -o ackermann ackermann.c && time ./ackermann

#include <stdio.h>
 
int ackermann(int m, int n)
{
        if (!m) return n + 1;
        if (!n) return ackermann(m - 1, 1);
        return ackermann(m - 1, ackermann(m, n - 1));
}
 
int main()
{
	printf("%d", ackermann(3, 11));
}

File: ackermann.rs
// rustc -C opt-level=3 ackermann.rs && time ./ackermann

fn ack(m: isize, n: isize) -> isize {
    if m == 0 {
        n + 1
    } else if n == 0 {
        ack(m - 1, 1)
    } else {
        ack(m - 1, ack(m, n - 1))
    }
}
 
fn main() {
    println!("{}", ack(3, 11));
}
